src/openms/source/ANALYSIS/ID/SiriusExportAlgorithm.cpp

```diff
@@ -0,0 +1,274 @@
+                ).withMinInt(1);
+
+      parameter(
+                  OpenMSName("precursor_mz_tolerance"),
+                  DefaultValue(10.0),
+                  Description("Tolerance window for precursor selection (Feature selection in regard to the precursor)")
+                );
+
+      parameter(
+                  OpenMSName("precursor_mz_tolerance_unit"),
+                  DefaultValue("ppm"),
+                  Description("Unit of the precursor_mz_tolerance")
+               ).withValidStrings({"Da", "ppm"});
+
+      parameter(
+                  OpenMSName("precursor_rt_tolerance"),
+                  DefaultValue(5.0),
+                  Description("Tolerance window (left and right) for precursor selection [seconds]")
+               );
+
+      parameter(
+                  OpenMSName("isotope_pattern_iterations"),
+                  DefaultValue(3),
+                  Description("Number of iterations that should be performed to extract the C13 isotope pattern. "
+                              "If no peak is found (C13 distance) the function will abort. "
+                              "Be careful with noisy data - since this can lead to wrong isotope patterns")
+                );
+
+      flag(
+            OpenMSName("feature_only"),
+            Description("Uses the feature information from in_featureinfo to reduce the search space to MS2 "
+                        "associated with a feature")
+          );
+
+      flag(
+            OpenMSName("no_masstrace_info_isotope_pattern"),
+            Description("Use this flag if the masstrace information from a feature should be discarded "
+                       "and the isotope_pattern_iterations should be used instead")
+          );
+    }
+
+
+    void SiriusExportAlgorithm::updateExistingParameter(const OpenMS::Param &param)
+    {
+      for (auto it = param.begin(); it != param.end(); ++it)
+      {
+        const std::string name = it.getName();
+        if (hasFullNameParameter(name))
+        {
+          vector<std::string> tags(it->tags.begin(), it->tags.end());
+          param_.setValue(name, it->value, it->description, tags);
+        }
+      }
+    }
+
+    bool SiriusExportAlgorithm::hasFullNameParameter(const OpenMS::String &name) const
+    {
+      //alternative: return std::any_of(param_.begin(), param_.end(), [&](const auto& it) { return it.name == name; });
+      for (auto it = param_.begin(); it != param_.end(); ++it)
+      {
+        if (it.getName() == name)
+        {
+          return true;
+        }
+      }
+      return false;
+    }
+
+
+    // ################
+    // Algorithm
+    // ################
+    void SiriusExportAlgorithm::preprocessingSirius(const String& featureinfo,
+                                                     const MSExperiment& spectra,
+                                                     FeatureMapping::FeatureMappingInfo& fm_info,
```

Comment:
Consider using more descriptive variable names. For example, `fm_info` could be renamed to something more descriptive like `feature_mapping_info`. This would improve code readability and maintainability. [medium]